### PR TITLE
docs: page about client-side configuration + per-host-overrides

### PR DESCRIPTION
**Pull request by jrudolph**
Refs #3212

### CODE REVIEW

File: docs/src/main/paradox/client-side/configuration.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * `akka.http.host-connection-pool`: pool settings
+ 
+## Basic Client Settings
+
+These settings influence the basic library behavior for each HTTP connection. When changed in the application's
+`application.conf` they globally affect the default client behavior.
+
+Basic client settings can be overridden in multiple ways:
+
+ * by passing custom @apidoc[ClientConnectionSettings] instances to APIs in @scala[@apidoc[Http$]]@java[@apidoc[Http]]
```
Comment (raboof) [inline docs/src/main/paradox/client-side/configuration.md:15]: The `$` is automatically chopped off for the java apidoc, so this should be equivalent:
```suggestion
 * by passing custom @apidoc[ClientConnectionSettings] instances to APIs in @apidoc[Http$]
```

File: docs/src/main/paradox/client-side/configuration.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+Pool settings can be overridden on a [per-target-host](#per-host-overrides) basis.
+
+@@snip [reference.conf]($akka-http$/akka-http-core/src/main/resources/reference.conf) { #pool-settings }
+
+## Per Host Overrides
+
+Settings can be overridden on a per-host basis by creating a list of `host-patterns` together with overridden settings
+in the `akka.http.host-connection-pool.per-host-override` setting.
+
+Note that only the first matching entry is selected and used even if multiple entries would match.
```
Comment (raboof) [inline docs/src/main/paradox/client-side/configuration.md:41]: :+1:

File: docs/src/main/paradox/client-side/configuration.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+Basic client settings can be overridden in multiple ways:
+
+ * by passing custom @apidoc[ClientConnectionSettings] instances to APIs in @scala[@apidoc[Http$]]@java[@apidoc[Http]]
+ * by overriding settings in `akka.http.host-connection-pool.client`, these overrides will take effect whenever a pool is used
+   like with `Http().singleRequest`
+ * by putting custom @apidoc[ClientConnectionSettings] into @apidoc[ConnectionPoolSettings] and passing those to APIs in `Http`
+ * by using [per-host overrides](#per-host-overrides)
+
+@@snip [reference.conf]($akka-http$/akka-http-core/src/main/resources/reference.conf) { #client-settings }
```
Comment (raboof) [inline docs/src/main/paradox/client-side/configuration.md:21]: Paradox should already understand `/` as the project root, so the `$akka-http$` prefix is no longer needed (and makes the link work when viewing the markdown in the github ui)

```suggestion
@@snip [reference.conf](/akka-http-core/src/main/resources/reference.conf) { #client-settings }
```

File: docs/src/main/paradox/client-side/configuration.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+Pool settings influence the behavior of client connection pools as used with APIs like `Http.singleRequest`
+(see @ref[request-level](request-level.md) and @ref[host-level](host-level.md)).
+
+This includes the amount of total concurrent connections a pool should open to a target host and other settings.
+These settings include a (by default empty) section `client` that can be used to override basic client settings
+when used in the context of a pool.
+
+Pool settings can be overridden on a [per-target-host](#per-host-overrides) basis.
+
+@@snip [reference.conf]($akka-http$/akka-http-core/src/main/resources/reference.conf) { #pool-settings }
```
Comment (raboof) [inline docs/src/main/paradox/client-side/configuration.md:34]:
```suggestion
@@snip [reference.conf](/akka-http-core/src/main/resources/reference.conf) { #pool-settings }
```

File: docs/src/main/paradox/client-side/configuration.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+## Per Host Overrides
+
+Settings can be overridden on a per-host basis by creating a list of `host-patterns` together with overridden settings
+in the `akka.http.host-connection-pool.per-host-override` setting.
+
+Note that only the first matching entry is selected and used even if multiple entries would match.
+
+@@snip [reference.conf]($akka-http$/akka-http-core/src/main/resources/reference.conf) { #per-host-overrides }
```
Comment (raboof) [inline docs/src/main/paradox/client-side/configuration.md:43]:
```suggestion
@@snip [reference.conf](/akka-http-core/src/main/resources/reference.conf) { #per-host-overrides }
```

File: docs/src/main/paradox/client-side/configuration.md (hunk 1 above)
Comment (jrudolph) [reply docs/src/main/paradox/client-side/configuration.md:15]: Ah, cool, copied that from somewhere else...

File: docs/src/main/paradox/client-side/configuration.md (hunk 3 above)
Comment (jrudolph) [reply docs/src/main/paradox/client-side/configuration.md:21]: Nice as well, also copied from somewhere else :)

Comment (jrudolph) [reply docs/src/main/paradox/client-side/configuration.md]: Fixed all of those in #3339.

**Pull request merged**